packages/main/src/RPA/MSGraph.py

```diff
@@ -0,0 +1,956 @@
+        # pylint: disable=anomalous-backslash-in-string
+        """Uploads a file to the specified OneDrive folder.
+
+        The uploaded file is returned as a DriveItem object and it has
+        additional properties that can be accessed with dot-notation, see
+        \`List Files In Onedrive Folder\` for details.
+
+        :param file_path: Path of the local file being uploaded.
+        :param target_folder: Path of the folder in OneDrive.
+        :param resource: Name of the resource if not using default.
+        :param drive_id: Drive ID if not using default.
+
+        .. code-block: robotframework
+
+            *** Tasks ***
+            Upload file
+                ${file}=    Upload File To Onedrive
+                ...    /path/to/file.txt
+                ...    /path/to/folder
+        """  # noqa: W605
+        self._require_authentication()
+        drive_instance = self._get_drive_instance(resource, drive_id)
+        folder = self._get_folder_instance(drive_instance, target_folder)
+        return folder.upload_file(item=file_path)
+
+    @keyword
+    def get_sharepoint_site(
+        self, *args: str, resource: Optional[str] = ""
+    ) -> sharepoint.Site:
+        """Returns a SharePoint site.
+
+        :param args: It accepts multiple ways of retrieving a site. See below.
+
+         get_site(host_name): the host_name e.g. 'contoso.sharepoint.com'
+         or 'root'.
+
+         get_site(site_id): the site_id is a comma separated string of
+         (host_name, site_collection_id, site_id).
+
+         get_site(host_name, path_to_site): host_name e.g. 'contoso.
+         sharepoint.com' and path_to_site is a url path (with a leading slash).
+
+         get_site(host_name, site_collection_id, site_id): a collection of
+         (host_name, site_collection_id, site_id).
+
+        :param resource: Name of the resource if not using default.
+        :return: SharePoint Site instance.
+
+        The return is of type Site and it has additional properties
+        that can be accessed with dot-notation. See examples below.
+
+        .. code-block: robotframework
+
+            *** Tasks ***
+            Get site
+                ${site}=    Get Sharepoint Site    contoso.sharepoint.com
+                Log    ${site.name}
+                Log    ${site.display_name}
+                Log    ${site.description}
+                Log    ${site.web_url}
+                Log    ${site.object_id}
+        """
+        self._require_authentication()
+        sp = self.client.sharepoint(resource=resource)
+        return sp.get_site(*args)
+
+    @keyword
+    def get_items_from_sharepoint_list(
+        self,
+        list_name: str,
+        site: sharepoint.Site,
+    ) -> DataTable:
+        # pylint: disable=anomalous-backslash-in-string
+        """Returns the items on a SharePoint list. The list is found
+        by it's display name.
+
+        This keyword tries to return the SharePoint list it as a table
```

Comment:
if in scope can be removed as they are in the same scope

Reply:
Agreed